code/lightna/magento-os-frontend/js/common/AddToCart.js

```diff
@@ -0,0 +1,80 @@
+import { Cookie } from '../lib/Cookie';
+import { UserInput } from '../lib/UserInput';
+import request from '../lib/HttpClient';
+import { $, $$ } from "../lib/utils";
+import { PageMessage } from './PageMessage';
+
+export class AddToCart {
+    components = [];
+    addToCartUrl = '/checkout/cart/add';
+    classes = {
+        loading: 'loading',
+        disabled: 'btn-disabled',
+    };
+
+    constructor() {
+        this.components = $$('.cjs-add-to-cart');
+        this.bindEvents();
+    }
+
+    bindEvents() {
+        this.components.foreach((i, component) => {
```

Comment:
I see we need a new ticket after merging it. This class works with collection of components.. cleaner will be having class instance per component + auto init. cjs class can contain class name cjs-AddToCart

Reply:
is it a request for changes or a request for ticket?

Reply:
for ticket, to merge this one faster :)